packages/dds/tree/src/util/randomSequenceGenerator.ts

```diff
@@ -0,0 +1,107 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { makeRandom } from "@fluid-internal/stochastic-test-utils";
+import { AnchorSet, Delta, FieldKey, UpPath } from "../tree";
+import { SequenceEditBuilder, singleTextCursor } from "../feature-libraries";
+import { jsonString } from "../domains";
+import { Transposed as T } from "../changeset";
+import { ChangeRebaser } from "../rebase";
+import { brand } from ".";
+
+// generate random UpPath objects
+export function generateRandomUpPaths(
+    parentKey: FieldKey, seed: number, maxUpPaths: number): Set<UpPath> {
```

Comment:
Rather than take a `maxUpPaths` I think it might be simpler to generate a single path and let the caller decide how many times to call the function. This means you'll be returning `UpPath` instead of `Set<UpPath>`.

It would be good to have a cap for the depth of the tree and a `maxIndex`.